### ISSUE TITLE
Training Code and probelm about C.TRACK.OFFSET

### ISSUE THREAD

**Issue by RaymondCover** [ohhhyeahhh/SiamGAT#4]
Excellent job, I want to know would the training code be available. And I also find that in siamgat_tracker.py  def accurate_location(),  dist = cfg.TRACK.OFFSET = 45, how do you calculate it? Thanks for your help.

**Comment by twotwo2** [comment]
We will release the training code as soon as possible. As for convolution/pooling layer, the coordinate mapping can be expressed as $$pos_i = stride_i * pos_{i+1} + ((kernel_i - 1) / 2 - padding)$$. Combined network structure of modified googlenet, the offset of the feature map to the original image is 45.

**Comment by twotwo2** [comment]
We will release the training code as soon as possible. As for convolution/pooling layer, the coordinate mapping can be expressed as pos_ = stride * pos + ((kernel - 1) / 2 - padding). Combined network structure of modified googlenet, the offset of the feature map to the original image is 45.

**Comment by RaymondCover** [comment]
I still have some problem about cfg.TRACK.OFFSET = 45, I train a model used resnet50 which used in SiamCAR, the score size is 31, but I got a very poor performance. Could you please offer the  cfg.TRACK.OFFSET for resnet50. And will you offer the results used resnet50 as the backbone? Thanks

**Comment by huchenjie339** [comment]
@RaymondCover 朋友，知道offset怎么算的吗？

**Comment by STQ-AmadeusUser** [comment]
Hello! I have figured out how to compute cfg.BACKBONE.OFFSET(=13 in /pysot/core/config.py)  by using the equation "pos_ = stride * pos + ((kernel - 1) / 2 - padding)". However, i failed understanding the cfg.TRACK.OFFSET(=45 in config.yaml). Since the backbones used in training code and testing code are both googlenet(inception v3), could you explain why the two offsets are different? Actually, if i change cfg.BACKBONE.OFFSET to 45 or cfg.TRACK.OFFSET to 13, poor performance will appear when tracking. Is the parameter 45 actually a hyper-parameter which can be tuned? Hope for your reply! Thanks! @twotwo2

**Comment by twotwo2** [comment]
@STQ-AmadeusUser 45 is not a hyper-parameter. _cfg.BACKBONE.OFFSET=13_ corresponding template branch, _cfg.TRACK.OFFSET=45_ corresponding search branch. To eliminate the impact of padding in backbone, we crop the boundary region on the search feature (crop_pad=4), so the offset is 13+crop_pad*stride=45.
The cropping code is in [googlenet](https://github.com/ohhhyeahhh/SiamGAT/blob/main/pysot/models/backbone/googlenet.py)
`            crop_pad = cfg.BACKBONE.CROP_PAD
            x = x[:, :, crop_pad:-crop_pad, crop_pad:-crop_pad]`

**Comment by STQ-AmadeusUser** [comment]
Thanks for your reply! It's very helpful for me!
